BundlePublisher exec plugin (#5939)
spiffe/spire: doc/plugin_server_bundlepublisher_exec.md

# Server plugin: BundlePublisher "exec"

> The `exec` plugin executes the specified command giving it the current trust bundle of the server
> on its stdin

Reviewer: ```suggestion
The `exec` plugin executes the specified command and passes the current trust bundle of the server
on its stdin
```

# Server plugin: BundlePublisher "exec"

The `exec` plugin executes the specified command and passes the current trust bundle of the server
on its stdin

The plugin accepts the following configuration options:

| Configuration     | Description                                                                                                                                                    | Required                                                               | Default                                             |
|-------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------|-----------------------------------------------------|

> | cmd               | Command to run                                                                                                                                                 | Yes.                                                                   |                                                     |
> | args              | Arguments to the command to run                                                                                                                                | No.                                                                    |                                                     |
> | format            | Format in which the trust bundle is stored, &lt;spiffe &vert; jwks &vert; pem&gt;. See [Supported bundle formats](#supported-bundle-formats) for more details. | No.                                                                    | spiffe                                              |

Reviewer: ```suggestion
| cmd               | Command to run                                                                                                                                                 | Yes                                                                    |                                                     |
| args              | Arguments to the command to run                                                                                                                                | No                                                                     |                                                     |
| format            | Format in which the trust bundle is stored, &lt;spiffe &vert; jwks &vert; pem&gt;. See [Supported bundle formats](#supported-bundle-formats) for more details. | No                                                                     | spiffe                                              |
```

## Supported bundle formats

The following bundle formats are supported:

### SPIFFE format

The trust bundle is represented as an RFC 7517 compliant JWK Set, with the specific parameters defined in the [SPIFFE Trust Domain and Bundle specification](https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format). Both the JWT authorities and the X.509 authorities are included.

### JWKS format

The trust bundle is encoded as an RFC 7517 compliant JWK Set, omitting SPIFFE-specific parameters. Both the JWT authorities and the X.509 authorities are included.

### PEM format

The trust bundle is formatted using PEM encoding. Only the X.509 authorities are included.

> ## Sample configuration

Author: It should work on all platforms.

I can take a rough stab, but don't have a windows box to test on. Do you?
